src/api/profile.ts

```diff
@@ -137,3 +137,72 @@ export const updatePersonalizedSchedule = async ({
     });
   });
 };
+
+export interface updateTalkShowStudioExperienceProps {
+  venueId: string;
+  userId: string;
+  experience: TalkShowStudioExperience;
+}
+
+export const updateTalkShowStudioExperience = async ({
+  venueId,
+  userId,
+  experience,
+}: updateTalkShowStudioExperienceProps) => {
+  const userProfileRef = getUserRef(userId);
+
+  const userData = (await userProfileRef.get()).data();
+
+  const newData = {
+    [`data.${venueId}`]: { ...userData?.data?.[venueId], ...experience },
+  };
```

Comment:
No need to fetch the data here
Any reason not to just replace it with new experience?

Reply:
@mike-lvov Why do you say no fetching? I'm not sure if that is correct/ideal?

Reply:
Though i'm also not sure I understand why we're doing a lot of what was done here either.